File: Contracts/Dto/Authentication/Register/RegisterRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace Contracts.Dto.Authentication.Register;

public class RegisterRequestDto
{
    [Required]
    [Display(Name = "Last name")]
    public string LastName { get; set; } = default!;

    [Required]
    [Display(Name = "First name")]
    public string FirstName { get; set; } = default!;

    [Required]
    [Display(Name = "Nickname")]
    public string UserName { get; set; } = default!;

    [Required]
    [Display(Name = "Email")]
    [EmailAddress]
    public string Email { get; set; } = default!;

    [Required]
    [Display(Name = "Password")]
    [DataType(DataType.Password)]
    public string Password { get; set; } = default!;

    [Required]
    [Compare("Password", ErrorMessage = "Пароли не совпадают")]
    [DataType(DataType.Password)]
    [Display(Name = "ConfirmPassword")]
    public string ConfirmPassword { get; set; } = default!;
    
    [Required]
    [DataType(DataType.Date)]
    [Display(Name = "Date of birth")]
    public DateTime DateOfBirth { get; set; } = default!;

    [Required]
    public double Latitude { get; set; } = default!;
    
    [Required]
    public double Longitude { get; set; } = default!;
    
    [Required] [Display(Name = "Gender")] 
    public string Gender { get; set; } = default!;

    [Display(Name = "Tell about yourself")]
    public string About { get; set; } = default!;
}

File: mobile/lib/Components/server/dto/register/register_request_dto.dart
class RegisterRequestDto {

  final String  lastName;
  final String  firstName;
  final String  userName;
  final String  email;
  final String  password;
  final String  confirmPassword;
  final String  dateOfBirth;
  final double  latitude;
  final double  longitude;
  final String  gender;
  final String  about;

  RegisterRequestDto(
      this.lastName,
      this.firstName,
      this.userName,
      this.email,
      this.password,
      this.confirmPassword,
      this.dateOfBirth,
      this.latitude,
      this.longitude,
      this.gender,
      this.about);
}